unittest/CMakeLists.txt

```diff
@@ -31,13 +31,15 @@ target_link_libraries(ofdm_mem m)
 
 add_library(function_trace STATIC ../unittest/function_trace.c)
 
+if (NOT APPLE) 
```

Comment:
Moving
``` 
target_link_libraries(ofdm_stack function_trace m -no-pie "-Wl,-Map=ofdm_stack.map")
```
into the "then" and
```
target_link_libraries(ofdm_stack function_trace m -no-pie ")
```
into "else" instead of wrapping it in the NOT APPLE if keeps it working for MacOS X.

Reply:
Updated. Thanks!

Reply:
This doesn't work for me. -no-pie doesn't exist for clang.

Reply:
I should note that my compiler is Apple Clang 7.3.0, not sure if the option is available in newer versions. Going back to the if (NOT APPLE) directive made this repo build again for me.

Reply:
On my 2017 MBP, FWIW:

MacBook-Pro:build_osx mooneer$ cc -v
Apple LLVM version 10.0.1 (clang-1001.0.46.4)
Target: x86_64-apple-darwin18.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
MacBook-Pro:build_osx mooneer$ 

Also, my version prints a message indicating that -no-pie isn't used:

[ 69%] Linking C executable ofdm_stack
cd /Users/mooneer/devel/freedv-gui/codec2/build_osx/unittest && /opt/local/bin/cmake -E cmake_link_script CMakeFiles/ofdm_stack.dir/link.txt --verbose=1
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc  -Wall -Wno-strict-overflow -std=gnu11 -fPIC -g -O2 -DDUMP -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names  CMakeFiles/ofdm_stack.dir/ofdm_stack.c.o CMakeFiles/ofdm_stack.dir/__/src/ofdm.c.o CMakeFiles/ofdm_stack.dir/__/src/octave.c.o CMakeFiles/ofdm_stack.dir/__/src/kiss_fft.c.o CMakeFiles/ofdm_stack.dir/__/src/modem_probe.c.o CMakeFiles/ofdm_stack.dir/__/src/mpdecode_core.c.o CMakeFiles/ofdm_stack.dir/__/src/phi0.c.o CMakeFiles/ofdm_stack.dir/__/src/filter.c.o  -o ofdm_stack libfunction_trace.a -lm -no-pie 
clang: warning: argument unused during compilation: '-nopie' [-Wunused-command-line-argument]
[ 72%] Built target ofdm_stack

Reply:
I guess they added support in for newer versions of clang, at least to note it's not used.

Still, maybe just disable it for APPLE entirely... unless there are actually developers who want to develop for the SM1000 under OSX? (and good luck to them...)